src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -436,13 +436,15 @@ private boolean quickSetSolid(final TransactionViewModel transactionViewModel) t
      * @throws Exception if we encounter an error while requesting a transaction
      */
     private boolean checkApproovee(TransactionViewModel approovee) throws Exception {
-        if(approovee.getType() == PREFILLED_SLOT) {
```

Comment:
this order swap! perfect -- the NULL_HASH also returns `PREFILLED_SLOT` so up until this fix it would not mark TXs pointing to Genesis as solid.